src/opera/pge/cslc_s1/cslc_s1_pge.py

```diff
@@ -188,20 +227,51 @@ def _cslc_filename(self, inter_filename):
 
         production_time = get_time_for_filename(self.production_datetime)
 
-        cslc_file_components = (
+        cslc_filename = (
             f"{core_filename}-{sensor}_{mode}_{burst_id}_{pol}_"
             f"{acquisition_time}Z_{product_version}_{production_time}Z"
         )
 
-        return cslc_file_components
+        # Cache the file name for this burst ID, so it can be used with the
+        # ISO metadata later
+        self._burst_filename_cache[burst_id] = cslc_filename
+
+        return cslc_filename
+
+    def _h5_filename(self, inter_filename):
+        """
+        Returns the file name to use for HDF5 products produced by the CSLC-S1 PGE.
+
+        The HDF5 filename for the CSLC-S1 PGE consists of:
+
+            <CSLC filename>.h5
+
+        Where <CSLC filename> is returned by CslcS1PostProcessorMixin._cslc_filename()
+
+        Parameters
+        ----------
+        inter_filename : str
+            The intermediate filename of the output HDF5 to generate a
+            filename for. This parameter may be used to inspect the file in order
+            to derive any necessary components of the returned filename.
+
```

Comment:
Do you mean this parameter can be searched for within the file to find any necessary components related to the returned file?  It's a little unclear to me.

Reply:
Since `inter_filename` is a path to the file to be renamed, it can be used to read the file contents from disk to extract any metadata needed to derive portions of the final filename. In certain instances, the file path itself could have some useful information in it as well, such as the burst ID for the product.